ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java

```diff
@@ -501,6 +506,74 @@ public void getBlobSubResourceTest()
     }
   }
 
+  /**
+   * This tests the construction of {@link GetBlobOptions} objects with various range and sub-resource settings.
+   * @throws RestServiceException
+   */
+  @Test
+  public void buildGetBlobOptionsTest()
+      throws RestServiceException {
+    // no range
+    doBuildGetBlobOptionsTest(null, null, true, true);
+    // valid ranges
+    doBuildGetBlobOptionsTest("bytes=0-7", ByteRange.fromOffsetRange(0, 7), true, false);
+    doBuildGetBlobOptionsTest("bytes=234-56679090", ByteRange.fromOffsetRange(234, 56679090), true, false);
+    doBuildGetBlobOptionsTest("bytes=1-", ByteRange.fromStartOffset(1), true, false);
+    doBuildGetBlobOptionsTest("bytes=12345678-", ByteRange.fromStartOffset(12345678), true, false);
+    doBuildGetBlobOptionsTest("bytes=-8", ByteRange.fromLastNBytes(8), true, false);
+    doBuildGetBlobOptionsTest("bytes=-123456789", ByteRange.fromLastNBytes(123456789), true, false);
+    // bad ranges
+    doBuildGetBlobOptionsTest("bytes=0-abcd", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=0as23-44444444", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=22-7777777777777777777777777777777777777777777", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=22--53", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=223-34", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=-34ab", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=--12", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=-12-", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=12ab-", null, false, false);
+    doBuildGetBlobOptionsTest("bytes=---", null, false, false);
+    doBuildGetBlobOptionsTest("btes=3-5", null, false, false);
+  }
+
+  private void doBuildGetBlobOptionsTest(String rangeHeader, ByteRange expectedRange,
+      boolean shouldSucceedWithoutSubResource, boolean shouldSucceedWithSubResource)
+      throws RestServiceException {
+    Map<String, Object> args = new HashMap<>();
+    if (rangeHeader != null) {
+      args.put(RestUtils.Headers.RANGE, rangeHeader);
+    }
+    doBuildGetBlobOptionsTestForSubResource(args, null, expectedRange, GetBlobOptions.OperationType.All,
+        shouldSucceedWithoutSubResource);
+    for (RestUtils.SubResource subResource : RestUtils.SubResource.values()) {
+      doBuildGetBlobOptionsTestForSubResource(args, subResource, expectedRange, GetBlobOptions.OperationType.BlobInfo,
```

Comment:
Do you mean that I should change the method signature such that `subResource` is the fourth argument?